### PR TITLE
kubeadm: add better test coverage to token.go

### PR DESCRIPTION
**Pull request by neolit123**
**What this PR does / why we need it**:
a PR for adding some more tests in `kubeadm/cmd` for `token.go`.

some areas of the `token.go` like listing, creating and deleting tokens can present challenges.
coverage was increased to around 87%.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

please, link issue # if you know of such.

**Special notes for your reviewer**:
none

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by neolit123** [comment]
updated - waiting on a review.
thanks.

**Comment by neolit123** [comment]
`verify-bazel` doesn't fail locally. could someone please explain why `pull-kubernetes-bazel-test` is failing here?

**Comment by neolit123** [comment]
thanks for having a look,

> Signed-off-by in every commit message is not needed.

noted.

> maybe better squash or split this PR for smaller ones. no sense of having so many small commits.

habit from other projects. i will squash everything in one commit. mostly attempting to help the reviewer, but these smaller commits are actually harder to rebase in case of conflicts. so :+1:

File: cmd/kubeadm/app/cmd/token_test.go [hunk 1]
```diff
@@ -142,3 +195,335 @@ func TestRunCreateToken(t *testing.T) {
...
+
+	tmpDir, err := ioutil.TempDir("", "kubeadm-token-test")
+	if err != nil {
+		t.Errorf("Unable to create temporary directory: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+	fullPath := filepath.Join(tmpDir, testConfigFile)
+
+	// test dryRun = false on a non-exisiting file
+	getClientset(fullPath, false)
```
Comment (kad) [inline cmd/kubeadm/app/cmd/token_test.go]: you probably need to check that returned error is as expected ?

File: cmd/kubeadm/app/cmd/token_test.go [hunk 2]
```diff
@@ -142,3 +195,335 @@ func TestRunCreateToken(t *testing.T) {
...
+
+	// test dryRun = true on a non-exisiting file
+	getClientset(fullPath, true)
```
Comment (kad) [inline cmd/kubeadm/app/cmd/token_test.go]: same here, check results ?

**Review by kad — COMMENTED** [review]
Please check also failure in TestRunCreateToken test case.

File: cmd/kubeadm/app/cmd/token_test.go (hunk 2 above)
Comment (neolit123) [reply cmd/kubeadm/app/cmd/token_test.go]: @kad updated with the two requests above.

Comment (kad) [reply cmd/kubeadm/app/cmd/token_test.go]: Tests seems to be failing still:
```
	token_test.go:194: Test case invalid: print join command: RunCreateToken expected error: true, saw: false
```

Comment (neolit123) [reply cmd/kubeadm/app/cmd/token_test.go]: should be fixed now.

**Comment by timothysc** [comment]
@neolit123 looks like bazel is not happy, could you rebase this PR and call ./hack/update-bazel.sh

**Comment by neolit123** [comment]
@timothysc 

seems that mergebot merged this PR commit, but the submit queue for this PR is still open.
`https://github.com/kubernetes/kubernetes/commit/a62a157c4883852472dc0ad7b8fcb506a0095672`

should i close this?

**Comment by neolit123** [comment]
the bazel build passed.

**Pull request merged**